File: src/main/java/burrows/apps/example/gif/di/module/RiffsyModule.java
package burrows.apps.example.gif.di.module;


import burrows.apps.example.gif.di.scope.PerActivity;
import burrows.apps.example.gif.rest.service.RiffsyRepository;
import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
@Module
public class RiffsyModule {
  @Provides @PerActivity protected RiffsyRepository provideRiffsyService(Retrofit.Builder retrofit) {
    return new RiffsyRepository(retrofit);
  }
}

File: src/main/java/burrows/apps/example/gif/di/module/RiffsyModule.kt
package burrows.apps.example.gif.di.module

import burrows.apps.example.gif.BuildConfig
import burrows.apps.example.gif.data.RiffsyApiClient
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
class RiffsyModule(var baseUrl: String = BuildConfig.BASE_URL) {
  @Provides fun providesRiffsyApi(retrofit: Retrofit.Builder)
    : RiffsyApiClient = retrofit
    .baseUrl(baseUrl)
    .build()
    .create(RiffsyApiClient::class.java)
}
